spec/core/ics-004-channel-and-packet-semantics/README-pubsub.md

```diff
@@ -0,0 +1,274 @@
+---
+ics: 4
+title: Broadcast Channel Semantics
+stage: draft
+category: IBC/TAO
+kind: instantiation
+requires: 2, 3, 5, 24
+author: Manuel Bravo <manuel@informal.systems>
+created: 2023-01-16
+modified: 2023-01-16
+---
+
+## Synopsis
+
+This standard document specifies the state machine handling logic of the broadcast channel abstraction.
+
+## Overview and Basic Concepts
+
+### Motivation
+
+TODO
+
+### Definitions
+
+`Connection` is as defined in [ICS 3](../ics-003-connection-semantics).
+
+`Port` and `authenticateCapability` are as defined in [ICS 5](../ics-005-port-allocation).
+
+`hash` is a generic collision-resistant hash function, the specifics of which must be agreed on by the modules utilising the channel. `hash` can be defined differently by different chains.
+
+`Identifier`, `get`, `set` and module-system related primitives are as defined in [ICS 24](../ics-024-host-requirements).
+
+## System Model and Properties
+
+### Desired Properties
+
+TODO
+
+## Technical Specification
+
+### General Design
+
+TODO
+
+During the handshake process, two ends of a channel come to agreement on a version bytestring associated
+with that channel. The contents of this version bytestring are and will remain opaque to the IBC core protocol.
+Host state machines MAY utilise the version data to indicate supported IBC/APP protocols, agree on packet
+encoding formats, or negotiate other channel-related metadata related to custom logic on top of IBC.
+
+Host state machines MAY also safely ignore the version data or specify an empty string.
+
+### Sub-protocols
+
+> Note: If the host state machine is utilising object capability authentication (see [ICS 005](../ics-005-port-allocation)), all functions utilising ports take an additional capability parameter.
+
+#### Opening a broadcast channel
+
+The `broadcastChanOpen` function is called by a module to open a broadcast channel. 
+
+```typescript
+function broadcastChanOpen(
+  order: ChannelOrder,
+  portIdentifier: Identifier,
+  version: string): CapabilityKey {
+    abortTransactionUnless(order !== ORDERED_ALLOW_TIMEOUT)
```

Comment:
The thing here is that the broadcaster does not expect any other chain to receive it, so timeouts do not apply: it could actually open it and start broadcasting packets before any other chain subscribes.

A broadcast channel can be ordered or unordered, so we could not just use the regular ordered channel or a new type.